src/commands/duel.ts

```diff
@@ -0,0 +1,287 @@
+      }
+
+      // Get the accepter from the DB. Create them if they don't exist yet.
+      let accepter = await this.client.duels.findUnique({
+        where: {
+          userName: accepterName,
+        },
+      })
+      if (!accepter) {
+        accepter = await this.client.duels.create({ data: { userName: accepterName } })
+      }
+
+      // Check if the accepter has recently lost and can't duel right now. Print their timeout.
+      if (accepter.lastLoss.getTime() + this.cooldown > Date.now()) {
+        const remaining = Math.ceil(Math.abs(Date.now() - (accepter.lastLoss.getTime() + this.cooldown)) / 1000)
+        await collectionInteraction.followUp({
+          content: `You have just duelled and lost. Please wait ${remaining} seconds before trying again.`,
+        })
+      } else if (!this.inProgress) {
+        // This case is not really supposed to happen because you should not be able to accept a duel after it has expired
+        // We are handling this anyways
+
+        // Check if there is no current duel
+        await collectionInteraction.followUp({
+          content: `There is no duel right now, it has probably expired. Use /duel to create one.`,
+        })
+        // Disable the button
+        const button = this.createButton(true)
+        const row = new MessageActionRow().addComponents(button)
+        await collectionInteraction.editReply({
+          components: [row],
+        })
+        return
+      } else {
+        // Disable duel
+        this.inProgress = false
+
+        // Disable the button
+        const button = this.createButton(true)
+        const row = new MessageActionRow().addComponents(button)
+        await collectionInteraction.editReply({
+          components: [row],
+        })
+
+        // Get and announce the winner
+        const accepterScore = this.getRandomScore()
+        const challengerScore = this.getRandomScore()
+        let winnerText = ''
+        if (challengerScore > accepterScore) {
+          await this.updateUserScore(challengerName, 'win')
+          await this.updateUserScore(accepterName, 'loss')
+
+          winnerText = `${challengerName} has won!`
+        } else if (accepterScore > challengerScore) {
+          await this.updateUserScore(challengerName, 'loss')
+          await this.updateUserScore(accepterName, 'win')
+
+          winnerText = `${accepterName} has won!`
+        } else {
+          await this.updateUserScore(challengerName, 'draw')
+          await this.updateUserScore(accepterName, 'draw')
+
+          winnerText = "It's a draw!"
+        }
+
+        await collectionInteraction.followUp({
+          content: `${accepterName} has rolled a ${accepterScore} and ${challengerName} has rolled a ${challengerScore}. ${winnerText}`,
+        })
+      }
+    })
+  }
+
+  @Slash('duelstats')
+  private async duelStats(interaction: CommandInteraction) {
+    await interaction.deferReply()
+
+    const userName = interaction.user.username
```

Comment:
This should probably be a user mention `interaction.user.toString()` or if we don't want to mention them their current nickname if it's set otherwise their username:
  `interaction.guild?.members.cache.find((member) => member.id === interaction.user.id)?.nickname ?? interaction.user.username`

Reply:
Probably should do the same above in Duel for the Challenger and Challengee